### PR TITLE
Skin padding implementation

### PR DESCRIPTION
**Pull request by qwertychouskie**
Needed for my new theme.  Tested with all existing themes, and all should be well.

## Agreement
```
By creating a pull request in stk-code, you hereby agree to dual-license your contribution as
GNU General Public License version 3 or any later version and
Mozilla Public License version 2 or any later version.

This includes your previous contribution(s) under the same name of contributor.

Keep the above statement in the pull request comment for agreement.

```

### CODE REVIEW

**Comment by CodingJellyfish** [comment]
Ok i'll close mine.
(Looks like it's Qwerty's biggest PR ever!)

**Comment by Benau** [comment]
image
now it has gaps between tabs

**Comment by qwertychouskie** [comment]
If the gaps bother you, they can be removed by setting a negative `h_margin`, e.g. `-8`.

**Comment by Benau** [comment]
if any negative values don't matter then it should not be a -8 (too arbitrary!), maybe all those padding values just need a fixed set of values for old skin

**Comment by Benau** [comment]
image
Still problemtic.... 
LHS your branch, RHS original, seems that your branch doesn't seem to use available space in a good way...

File: src/guiengine/screen_keyboard.cpp [hunk 1]
```diff
@@ -225,17 +223,21 @@ void ScreenKeyboard::createButtons()
...
         for (int j = 0; j < cols_num; j++)
         {
+            ButtonWidget* button = new ButtonWidget();
+
             float width = (float)total_width * layout_proportions[i][j] 
                                              / total_proportions - margin;
             
             char width_str[100];
-            sprintf(width_str, "%i", (int)roundf(width));
+            sprintf(width_str, "%i", (int)roundf(width / (SkinConfig::getHorizontalInnerPadding(button->getType(), button)+1.0f)));
```
Comment (auriamg) [inline src/guiengine/screen_keyboard.cpp]: why +1.0f ?

**Comment by Benau** [comment]
image
still bad space usage

File: src/guiengine/screen_keyboard.cpp (hunk 1 above)
Comment (Benau) [reply src/guiengine/screen_keyboard.cpp]: qwerty said in irc btw:
```
1.0f is the height of the button with no inner padding, so (SkinConfig::getHorizontalInnerPadding(button->getType(), button)+1.0f) is the real height of the button
```

**Pull request merged**